File: Dsa/weekly-practice/practiceday4.py



def AschiChar(char):
    for i in char:
        if "0"<=i<="9":
            continue
        if "a"<=i<="z":
            continue
        if "A"<=i<="Z":
            continue
        return False
    return True

def freqChar(arr):
    freq={}
    maxcount=0
    maxword=""
    for i in arr:
        if(i in freq):
            freq[i]+=1
        else:
            freq[i]=1
    for i in freq:
        if freq[i]>maxcount:
            maxcount=freq[i]
            maxword=i
    return maxword, maxcount

print(AschiChar("hfjsdf8483"))
print(Aschi("hfjsdf8483"))
print(freqChar("apple"))

File: Dsa/weekly-practice/practiceday4.js
function AschiChar(arr){
    for(i=0; i<arr.length; i++){
        let ch=arr[i]
        if(ch>="0" && ch<="9") continue
        if(ch>="a" && ch<="z") continue
        if(ch>="A" && ch<="Z") continue

        return false
    }
    return true

}

function newAschichar(arr){
    for(let i=0; i<arr.length; i++){
        let ch=arr[i]
        if((ch>="0" && ch<="9") || (ch>="A" && ch<="Z") || (ch>="a" && ch<="z")){
            return false
        }
    }
    return true
}

function freqchar(str){
    freq={}
    for(let ch of str){
        freq[ch]=(freq[ch]|| 0)+1
    }
    let maxcount=0
    let maxchar=""
    for(let ch in freq){
        if(freq[ch]>maxcount){
            maxcount=freq[ch]
            maxchar=ch
        }
    }
    return {maxchar, maxcount}
}
console.log(AschiChar("njdshsjkfh874534jhfjhsd"))
console.log(newAschichar("sdkfjhsdjfuiew237483$"))
console.log(freqchar("Apple Apple"))